server/http/csp/csp.go

```diff
@@ -0,0 +1,23 @@
+package csp
+
+import (
+	"io"
+	"net/http"
+
+	"github.com/buildbuddy-io/buildbuddy/server/util/alert"
+)
+
+// Nonce is the context key type for the per-request CSP nonce.
+// The associated value is always of type template.HTMLAttr.
+type Nonce struct{}
+
+const ReportingEndpoint = "/csp-report"
```

Comment:
it makes me a little nervous that anyone could `curl` this endpoint and trigger an alert for us - maybe just `log.CtxInfo()` for now?

Reply:
Updated to `log.CtxWarning`

Reply:
Warning-level logs are considered errors in our canarying process, and can trigger the apps to be rolled back if too many warnings are logged - we might want to use `Info` until we're more confident that there shouldn't be any CSP violations

Reply:
TIL, thanks!